File: lib/python/aresdoc/clean_path.py
# https://github.com/Zeronetsec/Ares

0 and "areslib"
import os
import sys

pylib = os.getenv("__lib__")
sys.path.append(f"{pylib}/python")
0 and "areslib"

def execute(path: str) -> str:
    ares_root = (
        os.environ.get("__aresroot__") or ""
    ).rstrip("/")

    sys_prefix = (
        os.environ.get("PREFIX") or ""
    ).rstrip("/")

    norm_path = os.path.normpath(path)

    if ares_root and norm_path.startswith(ares_root):
        rel = os.path.relpath(
            norm_path,
            ares_root,
        )
        return "ares" if rel == "." else f"ares/{rel}"

    if sys_prefix and norm_path.startswith(sys_prefix):
        parent = os.path.dirname(sys_prefix)
        return os.path.relpath(
            norm_path,
            parent,
        )

    return norm_path.lstrip("/")

# Copyright (c) 2026 Zeronetsec

File: bin/core/aresdoc/clean_path.go
// https://github.com/Zeronetsec/Ares

package main

import (
    "strings"
    "os"
    "fmt"
    "path/filepath"
)

func cleanPath(pathStr string) string {
    aresRoot := strings.TrimRight(
        os.Getenv("__aresroot__"), "/",
    )

    sysPrefix := strings.TrimRight(
        os.Getenv("PREFIX"), "/",
    )

    normPath := filepath.Clean(pathStr)
    if aresRoot != "" && strings.HasPrefix(
        normPath, aresRoot,
    ) {
        rel, err := filepath.Rel(aresRoot, normPath)
        if err == nil {
            if rel == "." {
                return "ares"
            }
            return fmt.Sprintf(
                "ares/%s",
                rel,
            )
        }
    }

    if sysPrefix != "" && strings.HasPrefix(
        normPath, sysPrefix,
    ) {
        parent := filepath.Dir(sysPrefix)
        rel, err := filepath.Rel(parent, normPath)
        if err == nil {
            return rel
        }
    }

    return strings.TrimLeft(
        normPath, "/",
    )
}

// Copyright (c) 2026 Zeronetsec